pyproject.toml

```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "agentops"
-version = "0.3.4"
+version = "0.4.0"
```

Comment:
are we sure we want to make this a change to 0.4? is a big refactor internally, but to the developer it's not a significant change as there are no breaking changes or major features added. i personally think this should just be a patch but open to discussion